### PR TITLE
gh #154 VTS - L3 dsCompisiteIn C-Test Spec and Code Implementation

### PR DESCRIPTION
**Pull request by thanushreevani**
updated ctest with test scenarios

### CODE REVIEW

File: src/test_l3_dsCompositeIn.c [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+    else {
+        UT_LOG_INFO("Result dsCompositeInRegisterConnectCB(CBFunc:[0x%0X]) dsError_t:[%s]",
+                        compositeInConnectCB, ut_control_GetMapString(dsError_mapTable, ret));
+    }
+
+    /* Register Signal change callback */
+    UT_LOG_INFO("Calling dsCompositeInRegisterSignalChangeCB(cbFun:[0x%0X])", compositeInSignalChangeCB);
+    ret = dsCompositeInRegisterSignalChangeCB(compositeInSignalChangeCB);
+    if (ret != dsERR_NONE) {
+        UT_LOG_ERROR("Failed dsCompositeInRegisterSignalChangeCB(cbFun:[0x%0X]) dsError_t:[%s]",
```
Comment (Anbukannadhasan) [inline src/test_l3_dsCompositeIn.c:275]: Consider using ASSERT. Change the print formats. Refer https://github.com/rdkcentral/rdk-halif-test-device_settings/pull/116. Please change through out the function

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l3_dsCompositeIn.c [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+* This test function is to scale video of selected CompositeIn port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 005@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [dsCompositeIn_L3_Low-Level_TestSpecification.md](../docs/pages/ds-compositeIn_L3_Low-Level_TestSpecification.md)
+*/
+void test_l3_CompositeIn_scale_video(void)
```
Comment (hari22yuva) [inline src/test_l3_dsCompositeIn.c:392]: scale is applied on port right 
so ask user to select the port

**Review by hari22yuva — COMMENTED** [review]
partial review

File: src/test_l3_dsCompositeIn.c [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+* @brief This test get number of input ports.
+*
+* This test function gets Composite Input ports on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 002@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [dsCompositeIn_L3_Low-Level_TestSpecification.md](../docs/pages/ds-compositeIn_L3_Low-Level_TestSpecification.md)
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l3_dsCompositeIn.c]: done

File: src/test_l3_dsCompositeIn.c [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+* This test function selects CompositeIn port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 004@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [dsCompositeIn_L3_Low-Level_TestSpecification.md](../docs/pages/ds-compositeIn_L3_Low-Level_TestSpecification.md)
+*/
+void test_l3_CompositeIn_select_port(void)
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l3_dsCompositeIn.c]: done

File: src/test_l3_dsCompositeIn.c (hunk 2 above)
Comment (thanushreevani) [reply src/test_l3_dsCompositeIn.c]: done

File: src/test_l3_dsCompositeIn.c [hunk 5]
```diff
@@ -0,0 +1,480 @@
...
+    UT_LOG_INFO("Calling dsCompositeInRegisterConnectCB(CBFunc:[0x%0X])", compositeInConnectCB);
+    ret = dsCompositeInRegisterConnectCB(compositeInConnectCB);
+    UT_LOG_INFO("Result dsCompositeInRegisterConnectCB(CBFunc:[0x%0X]) dsError_t:[%s]",
+                        compositeInConnectCB, UT_Control_GetMapString(dsError_mapTable, ret));
+    ASSERT(ret == dsERR_NONE);
+
+    /* Register Signal change callback */
+    UT_LOG_INFO("Calling dsCompositeInRegisterSignalChangeCB(cbFun:[0x%0X])", compositeInSignalChangeCB);
+    ret = dsCompositeInRegisterSignalChangeCB(compositeInSignalChangeCB);
+    UT_LOG_INFO("Result dsCompositeInRegisterSignalChangeCB(cbFun:[0x%0X]) dsError_t:[%s]",
```
Comment (Anbukannadhasan) [inline src/test_l3_dsCompositeIn.c:240]: cbFun , CBFunc we can have one common name. In some places, we are using CBFunc and some places named as  cbFun

File: src/test_l3_dsCompositeIn.c [hunk 6]
```diff
@@ -0,0 +1,480 @@
...
+    dsError_t ret = dsERR_NONE;
+
+    /* Initialize the CompositeIn Module */
+    UT_LOG_INFO("Calling dsCompositeInInit()");
+    ret = dsCompositeInInit(); 
+    UT_LOG_INFO("Result dsCompositeInInit() dsError_t:[%s]", UT_Control_GetMapString(dsError_mapTable, ret));
+    ASSERT(ret == dsERR_NONE);
+
+    /* Register connection status callback */
+    UT_LOG_INFO("Calling dsCompositeInRegisterConnectCB(CBFunc:[0x%0X])", compositeInConnectCB);
```
Comment (Anbukannadhasan) [inline src/test_l3_dsCompositeIn.c:231]: Add print fomat in the below way

 UT_LOG_INFO("Calling dsCompositeInRegisterConnectCB(IN:CBFunc:[0x%0X])", compositeInConnectCB);

consider in all places

File: src/test_l3_dsCompositeIn.c [hunk 7]
```diff
@@ -0,0 +1,480 @@
...
+*/
+void test_l3_CompositeIn_select_and_scale_video(void)
+{
+    gTestID = 3;
+    UT_LOG_INFO("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int32_t x = 0 , y = 0, width = 0 , height = 0;
+    dsCompositeInPort_t port = dsCOMPOSITE_IN_PORT_MAX;
+    int32_t select;
```
Comment (Anbukannadhasan) [inline src/test_l3_dsCompositeIn.c:306]: Initiaize with default values

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l3_dsCompositeIn.c (hunk 5 above)
Comment (thanushreevani) [reply src/test_l3_dsCompositeIn.c]: done

File: src/test_l3_dsCompositeIn.c (hunk 7 above)
Comment (thanushreevani) [reply src/test_l3_dsCompositeIn.c]: done

File: src/test_l3_dsCompositeIn.c (hunk 1 above)
Comment (thanushreevani) [reply src/test_l3_dsCompositeIn.c]: done

File: src/test_l3_dsCompositeIn.c [hunk 8]
```diff
@@ -0,0 +1,515 @@
...
+        else 
+        {
+            UT_LOG_ERROR("Invalid platform type: %s", gDeviceType);
+            return -1;
+        }
+    }
+
+    UT_add_test( pSuite, "Initialize CompositeIn" ,test_l3_CompositeIn_initialize );
+    UT_add_test( pSuite, "Get status of ports" ,test_l3_CompositeIn_get_status );
+    UT_add_test( pSuite, "select port" ,test_l3_CompositeIn_select_port );
```
Comment (Anbukannadhasan) [inline src/test_l3_dsCompositeIn.c:503]: select port - Add as "Select Port" instead.

**Review by Anbukannadhasan — COMMENTED** [review]
Added minor review comment

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Review by Anbukannadhasan — APPROVED** [review]
Approved

File: src/test_l3_dsCompositeIn.c (hunk 2 above)
Comment (hari22yuva) [reply src/test_l3_dsCompositeIn.c:349]: ok fine

**Review by hari22yuva — APPROVED** [review]
changes looks ok

**Pull request merged**